### PR TITLE
[make:schedule] Add transport name for attribute

### PR DESCRIPTION
**Pull request by NicolasJourdan**
Hey ! 👋 

I often use the command `make:schedule` and I wanted to add this new feature : Ask for the `name` used in the attribute `#[AsSchedule]`. 

## Without name 
Console : 
Capture d’écran 2024-06-01 à 11 49 48

Generated class : 

```php
<?php

namespace App\Scheduler;

use Symfony\Component\Scheduler\Attribute\AsSchedule;
use Symfony\Component\Scheduler\RecurringMessage;
use Symfony\Component\Scheduler\Schedule;
use Symfony\Component\Scheduler\ScheduleProviderInterface;
use Symfony\Contracts\Cache\CacheInterface;

#[AsSchedule]
final class MainSchedule implements ScheduleProviderInterface
{
    public function __construct(
        private CacheInterface $cache,
    ) {
    }

    public function getSchedule(): Schedule
    {
        return (new Schedule())
        ->add(
            // @TODO - Create a Message to schedule
            // RecurringMessage::every('1 hour', new App\Message\Message()),
        )
        ->stateful($this->cache)
        ;
    }
}
```

## With `dummy` name

Console : 
Capture d’écran 2024-06-01 à 11 50 01

Generated class : 

```php
<?php

namespace App\Scheduler;

use Symfony\Component\Scheduler\Attribute\AsSchedule;
use Symfony\Component\Scheduler\RecurringMessage;
use Symfony\Component\Scheduler\Schedule;
use Symfony\Component\Scheduler\ScheduleProviderInterface;
use Symfony\Contracts\Cache\CacheInterface;

#[AsSchedule('dummy')]
final class MainSchedule implements ScheduleProviderInterface
{
    public function __construct(
        private CacheInterface $cache,
    ) {
    }

    public function getSchedule(): Schedule
    {
        return (new Schedule())
        ->add(
            // @TODO - Create a Message to schedule
            // RecurringMessage::every('1 hour', new App\Message\Message()),
        )
        ->stateful($this->cache)
        ;
    }
}
```

### CODE REVIEW

**Review by mdoutreluingne — APPROVED** [review]
:+1:

File: src/Resources/skeleton/scheduler/Schedule.tpl.php [hunk 1]
```diff
@@ -4,7 +4,11 @@
 
 <?= $use_statements; ?>
 
+<?php if ($has_transport_name): ?>
+#[AsSchedule('<?= $transport_name; ?>')]
+<?php else: ?>
 #[AsSchedule]
+<?php endif ?>
```
Comment (jrushlow) [inline src/Resources/skeleton/scheduler/Schedule.tpl.php:11]:
```suggestion
#[AsSchedule<?php if ($has_transport_name): ?>('<?= $transport_name; ?>')<?php endif ?>]
```

**Review by jrushlow — CHANGES_REQUESTED** [review]
Awesome! One minor comment

File: src/Resources/skeleton/scheduler/Schedule.tpl.php (hunk 1 above)
Comment (NicolasJourdan) [reply src/Resources/skeleton/scheduler/Schedule.tpl.php]: Thanks ! 👌

**Review by jrushlow — APPROVED** [review]
Thank you @NicolasJourdan!

**Pull request merged**